WordPress-iOS-Shared/Core/WPNoResultsView.m

```diff
@@ -203,29 +203,40 @@ - (void)showInView:(UIView *)view {
 }
 
 - (void)centerInSuperview {
-    
+
     if (!self.superview) {
         return;
     }
-    
+
     // Center in superview
     CGRect frame = [self superview].frame;
-    
+
     // account for content insets of superview if it is a scrollview
     if ([self.superview.class isSubclassOfClass:[UIScrollView class]]) {
         UIScrollView *scrollView = (UIScrollView *)self.superview;
-        
+
         CGFloat verticalOffset = scrollView.contentInset.top + scrollView.contentInset.bottom;
         CGFloat horizontalOffset =  scrollView.contentInset.left + scrollView.contentInset.right;
-        
+
+        if ([self.superview isKindOfClass:[UITableView class]]) {
+            UITableView *tableView = (UITableView *)self.superview;
+            CGFloat headerHeight = tableView.tableHeaderView.hidden ? 0 : tableView.tableHeaderView.bounds.size.height;
+            CGFloat footerHeight = tableView.tableFooterView.hidden ? 0 : tableView.tableFooterView.bounds.size.height;
```

Comment:
I realize that when the `tableHeaderView` or `tableFooterView` is `nil` the value assigned will resolve to `0`, but the logic just seems a bit wonky, perhaps because there can be nils in the truth test and in the false condition.  Would it make sense to add a nil check as a part of the truth test to at least get the nil out of the false condition?  Something like:
`(tableView.tableHeaderView == nil || tableView.tableHeaderView.hidden)`
Thoughts?